Library/Homebrew/yard/templates/default/docstring/html/internal.erb

```diff
@@ -0,0 +1,5 @@
+<p class="note private">
+  <strong>This <%= object.type %> is part of an internal API.</strong>
+  This <%= object.type %> may only be used internally in repositories owned by <a href="https://github.com/Homebrew">Homebrew</a>.
```

Comment:
I'm not sure of the purpose of this distinction; if we're going to use it in Homebrew taps, why not allow it outside the Homebrew organisation with the usual deprecation cycle?

Reply:
Precisely to avoid (or at least shorten) this cycle when something is only used in our repos. To be clear, usage of `internal` APIs in casks/formulae should still not allowed even if we own the repo.

- `private`: no deprecation necessary
- `internal`: deprecation only necessary if migration is non-trivial
- `public`: deprecation always necessary

Reply:
I'm also thinking for `internal` APIs we don't need to have a “nice” API. I.e. in the case of `curl_output` we can just mark it `@api internal` and be done with it. With `@api public`, we should take more time to think about it: Do we really need `curl` with all its options? Does `curl_` need to be in the method name? Third parties should not need to know that `curl` is used under the hood. The public API should be as minimal as possible, i.e. `fetch(url)`.

Reply:
> I'm also thinking for `internal` APIs we don't need to have a “nice” API. I.e. in the case of `curl_output` we can just mark it `@api internal` and be done with it. With `@api public`, we should take more time to think about it: Do we really need `curl` with all its options? Does `curl_` need to be in the method name? Third parties should not need to know that `curl` is used under the hood. The public API should be as minimal as possible, i.e. `fetch(…)` or even `url … do`.

@reitermarkus Great argument, you've convinced me.